src/components/panel/panel.js

```diff
@@ -861,20 +837,15 @@ MdPanelRef.prototype.open = function() {
  * closed and animations finish.
  */
 MdPanelRef.prototype.close = function() {
-  // TODO(ErinCoughlan) - Cancel any in-progress actions.
-
   var self = this;
-  var closePromise = this.hide()
-      .then(function() {
-        return self.detach();
-      }, this._$q.reject)
-      .then(function() {
-        // TODO(ErinCoughlan) - Add destroy. This will make the code here
-        // different than just calling this.detach().
-        return self._$q.resolve(self);
-      }, this._$q.reject);
-
-  return closePromise;
+  return this._$q(function(resolve, reject) {
```

Comment:
Not true. Each `then( )` returns a new promise; so the chaining sequence is guaranteed.